tests/test_errors.py

```diff
@@ -0,0 +1,60 @@
+from io import BytesIO
+
+import pytest
+from lxml import etree
+
+from search.errors import NAMESPACE_MAP
+from tests.schema_utils import get_schema_file_path
+
+
+@pytest.mark.parametrize('status_code, name', [
+    (400, 'Bad Request'),
+    (401, 'Unauthorized'),
+    (403, 'Forbidden'),
+    (404, 'Not Found'),
+    (405, 'Method Not Allowed'),
+    (406, 'Not Acceptable'),
+    (408, 'Request Timeout'),
+    (409, 'Conflict'),
+    (410, 'Gone'),
+    (411, 'Length Required'),
+    # Custom response data and headers are removed for HTTP 412.
+    # See Flask issue 3093 for more information.
+    # (412, 'Precondition Failed'),
+    (413, 'Request Entity Too Large'),
+    (414, 'Request URI Too Long'),
+    (415, 'Unsupported Media Type'),
+    (416, 'Requested Range Not Satisfiable'),
+    (417, 'Expectation Failed'),
+    (418, 'I\'m a teapot'),
+    (428, 'Precondition Required'),
+    (429, 'Too Many Requests'),
+    (431, 'Request Header Fields Too Large'),
+    (500, 'Internal Server Error'),
+    (501, 'Not Implemented'),
+    (502, 'Bad Gateway'),
+    (503, 'Service Unavailable'),
+    (504, 'Gateway Timeout'),
+    (505, 'HTTP Version Not Supported'),
+    (None, 'Not Found'),  # Test /error uses default code when one is not supplied
+])
+def test_custom_error_handler(client, status_code, name) -> None:
+    url = '/error?code=%s' % status_code
+    if not status_code:
```

Comment:
I consider conditionals in test a smell since the increase the complexity of the test method. `test_custom_error_handler` already has a 404 case and it's testing its full response, so I'd remove the `None` case from the `parametrize` cases and transform it into a focused test that doesn't go through the XML version but only checks this "default" aspect:
```
def test_default_error_code(self):
    response = client.get('/error')
    assert response.status_code == 404
```

Reply:
updated.